DataFormats/SoATemplate/interface/SoACommon.h

```diff
@@ -813,4 +813,301 @@ SOA_HOST_ONLY std::ostream& operator<<(std::ostream& os, const SOA& soa) {
+  //   memcpy(destColumn.addr_, originColumn.addr_, originColumn.elements_ * sizeof(T));
+  // }
+
+  // template <typename T>
+  // void memcpyColumnFromView(cms::soa::SoAParametersImpl<cms::soa::SoAColumnType::eigen, T>& destColumn,
+  //                           const cms::soa::SoAParametersImpl<cms::soa::SoAColumnType::eigen, T>& originColumn) {
+  //   for (auto i = 0u; i < T::RowsAtCompileTime * T::ColsAtCompileTime; ++i) {
+  //     memcpy(destColumn.addr_ + i * destColumn.stride_,
+  //            originColumn.addr_,
+  //            sizeof(T::Scalar) * destColumn.stride_ * T::RowsAtCompileTime * T::ColsAtCompileTime);
+  //   }
+  // }
+
+  // TODO: This function can be used when the layout will contain ColumnTypes as members
+  // Helper functions for computing the location of columns in memory
+  // computeColumnPosition(BOOST_PP_CAT(typename Metadata::ParametersTypeOf_, NAME)(), BOOST_PP_CAT(NAME, _), _soa_impl_curMem, alignment, elements_);
+  //template <typename T>
+  //void computeColumnPosition(cms::soa::SoAParametersImpl<cms::soa::SoAColumnType::scalar, T>&& column,
+  //                           T*& columnAddr,
+  //                           std::byte*& currMem,
+  //                           cms::soa::byte_size_type alignment,
+  //                           cms::soa::size_type elements) {
+  //  using ColumnType = decltype(column);
+  //  //column = ColumnType(reinterpret_cast<T*>(currMem), elements);
+  //  columnAddr = reinterpret_cast<T*>(currMem);
+  //  currMem += cms::soa::alignSize(sizeof(T), alignment);
+  //}
+
+  //template <typename T>
+  //void computeColumnPosition(cms::soa::SoAParametersImpl<cms::soa::SoAColumnType::column, T>&& column,
+  //                           T*& columnAddr,
+  //                           std::byte*& currMem,
+  //                           cms::soa::byte_size_type alignment,
+  //                           cms::soa::size_type elements) {
+  //  using ColumnType = decltype(column);
+  //  //column = ColumnType(reinterpret_cast<T*>(currMem), elements);
+  //  columnAddr = reinterpret_cast<T*>(currMem);
+  //  currMem += cms::soa::alignSize(elements * sizeof(T), alignment);
+  //}
+
+  //template <typename T>
+  //void computeColumnPosition(cms::soa::SoAParametersImpl<cms::soa::SoAColumnType::eigen, T>&& column,
+  //                           typename T::Scalar*& columnAddr,
+  //                           std::byte*& currMem,
+  //                           cms::soa::byte_size_type alignment,
+  //                           cms::soa::size_type elements) {
+  //  using ColumnType = decltype(column);
+  //  //column = ColumnType(reinterpret_cast<T::Scalar*>(currMem),
+  //  columnAddr = reinterpret_cast<T::Scalar*>(currMem);
+  //  currMem += cms::soa::alignSize(elements * sizeof(typename T::Scalar), alignment) * T::RowsAtCompileTime *
+  //             T::ColsAtCompileTime;
+  //}
+}  // namespace cms::soa
+
```

Comment:
Can you remove this commented-out code from the PR, and add it later (uncommented) when possible and needed ?